src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -4919,8 +4919,10 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
       }
 
       def typedApply(tree: Apply) = tree match {
-        case Apply(Block(stats, expr), args) =>
-          typed1(atPos(tree.pos)(Block(stats, Apply(expr, args) setPos tree.pos.makeTransparent)), mode, pt)
+        case Apply(blk @ Block(stats, expr), args) =>
+          val ap1 = treeCopy.Apply(tree, expr, args).clearType().setPos(tree.pos.makeTransparent)
+          val blk1 = treeCopy.Block(blk, stats, ap1).clearType()
```

Comment:
Why did you add `clearType`?

Reply:
I wanted to pass on the attachment but I didn't want to change the current approach of re-typechecking the new `Apply` and `Block`

Reply:
Ah of course... Thanks